query-dsl/src/main/java/org/infinispan/query/dsl/QueryFactory.java

```diff
@@ -17,6 +17,12 @@
     */
    Query create(String queryString);
 
+   /**
+    * Creates a Query based on an Ickle query string
+    * @param queryMode the {@link IndexedQueryMode} dictating the indexed query execution mode if applicable.
+    */
+   Query create(String queryString, IndexedQueryMode queryMode);
```

Comment:
I'm looking for the way to build a QueryBuilder-DSL based query and also specify IndexedQueryMode but could not find it. Maybe I'm missing something.

And another thing, I'm not sure whether we need to specify IndexedQueryMode at query creation time; maybe it's better to specify it at execution time.

Reply:
I did not touch the DSL, I though it'd be in maintenance mode only?

WRT IndexedQueryMode at execution time, I've chosen at creation time because:
* It's how it's working now: SearchManager.getQuery vs SearchManager.getClusteredQuery
* I tried to avoid having to change the API everywhere, i.e., ```query.list(mode), query.iterator(mode), query.getResultSize(mode)``` for each of the query types